src/lib/assignment.ts

```diff
@@ -0,0 +1,127 @@
+import fs from 'fs'
+import path from 'path'
+import bent from 'bent'
+import FormData from 'form-data'
+import tar from 'tar'
+import glob from 'glob-promise'
+import YAML from 'yaml'
+import tools from './tools'
+import config from './config'
+import _ from 'lodash'
+
+type Yaml = {
+  assignment: string
+  paths: string[]
+  section: string | string[]
+}
+
+async function archiveTar(src: string): Promise<{ file: string; dir: string; }> {
+  const dir = await fs.promises.mkdtemp('codio_export')
+  const file = path.join(dir, 'project.tar.gz')
+  await tar.c(
+    {
+      gzip: true,
+      file,
+      cwd: src
+    },
+    ['./']
+  )
+  return {file, dir}
+}
+
+const sleep = async (seconds: number) => new Promise((resolve) => setTimeout(resolve, seconds * 1000))
+
+const getJson = bent('json')
+
+async function publishArchive (courseId: string, assignmentId:string, archivePath: string, changelog: string) {
```

Comment:
should return type be defined also, as in `archiveTar` function, something like `Promise<void>`?

Reply:
not necessary, but will update